### DISCUSSION TITLE
HOOK - Files are not removed from file system (disk)

### DISCUSSION THREAD

**Discussion by AlexVanK** [directus/directus#12438] [Archive]
Hello, I hope you guys can help me as this has been a major road block for months now, 
and I was hoping the issue would be addressed with future updates. 

In short: files are not removed from file-system (disk) when the deletion is triggered from a related parent, in a cascading way...

---

This is my collection structure:

Collection APPLICATION INFO - has a O2M field....
![image](https://user-images.githubusercontent.com/56647386/160451545-51868c47-1361-4d11-b90d-7a7bb833d7bf.png)

.. which is connected to Collection SUPPORTING DOCUMENTATION  that contains a file field:
![image](https://user-images.githubusercontent.com/56647386/160451693-3b6941bb-31ef-4b59-b47e-f9884ad4b7f9.png)

Logic:

Whenever an entry in APPLICATION INFO is deleted, all associated entries in SUPPORTING DOCUMENTATION should be deleted as well...
![image](https://user-images.githubusercontent.com/56647386/160451883-e833840e-1d8c-4e78-9e33-7a1c6c847c5b.png)

This works fine... However:

the files attached to  SUPPORTING DOCUMENTATION through the _file_ interface are not removed from the disk.

So I tried to implement that logic in a HOOK:
![image](https://user-images.githubusercontent.com/56647386/160452714-eafc4e7f-84d1-4ff7-8814-36b70865d417.png)
![image](https://user-images.githubusercontent.com/56647386/160453733-b9da4bdb-9db9-4246-9874-c238e7106525.png)

---

I found a previous post with a similar issue: https://github.com/directus/directus/discussions/8899

but in my case this has nothing to do with case sensitivity...

**Comment by AlexVanK** [comment]
Fixed this myself. Wasn't accessing the files correctly.

**Comment by NilsBaumgartner1994** [comment]
This does sadly not work on connected M2M tables

Forms --> Many Form Answers --> Many Files

**Reply by NilsBaumgartner1994** [reply]
Related to https://github.com/directus/directus/discussions/24663
@AlexVanK please consider giving the combined discussion an upvote in order to become implemented
